app/controllers/submissions_controller.rb

```diff
@@ -275,7 +275,17 @@ def score
 
     # Submit the score
     if @submission.users.map {|user| lti_outcome_service?(@submission.exercise, user, @submission.study_group_id) }.any?
-      transmit_lti_score
+      url = finalize_submission_path(@submission)
+      submit_info = transmit_lti_score
+      submit_info&.[](:status)&.each do |s|
+        client_socket&.send_data(JSON.dump({
+          cmd: :status,
+          status: s,
```

Comment:
Do you think it would be easier to send up to two distinct messages instead of one for LTI success / failure and on time / after the deadline? 

The LTI messages could be translated to the following LTI styles:
- success (all successful)
- warning (some failed, some worked)
- danger (all failed)

For the deadline, we could:
- not show any message if the deadline has been fulfilled
- an info if the deadline has passed (i.e., the score is reduced somehow). Here, I originally thought of having another warning (yellow), but this doesn't play nice with the LTI message styles (we can only display one warning at a time). Therefore, I think the info (blue) might be the second-best option currently.

Before changing anything here, do you think this change would make sense? Is it reducing the overall complexity / making the code easier to maintain (because those two information are no longer "merged") or do you have other ideas?

Reply:
Yes, probably you are right that this change would reduce some complexity. Then we only have three messages + sometimes an additional one and it also is easier to maintain the messages. 
I am just not sure what we should write in the messages if the full score is reached but after the deadline because then the points are reduced. Maybe we then write in the success message only that the exercise was solved completely without saying anything about the points and then in the additional message for the too late submission that the points are reduced?

Reply:
Yeah, I like that! 👍  This means, we are further decoupling _finishing_ the exercise (in terms of solving it to the satisfaction of the unit tests) and _scoring_ it (which could be influenced by the deadline).